src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,132 @@ map_injective f.injective u v
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons a p) H h := by
+  { refine walk.cons _ (p.transfer H _);
+    simp only [walk.edges_cons, list.mem_cons_iff, forall_eq_or_imp, mem_edge_set] at h,
+    exact h.1, exact h.2, }
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma transfer_id : p.transfer G (λ e ep, edges_subset_edge_set p ep) = p := by
+begin
+  induction p,
+  { simp only [transfer] },
+  { simp only [p_ih, transfer, eq_self_iff_true, heq_iff_eq, and_self] },
+end
+
+/-- The walk `p` transfered to the larger graph `H` -/
+abbreviation transfer_le (GH : G ≤ H) : H.walk u v :=
+p.transfer H (λ e ep, edge_set_mono GH (edges_subset_edge_set p ep))
+
+lemma transfer_eq_map_spanning_subgraphs (GH : G ≤ H) :
+  p.transfer H hp = p.map (simple_graph.hom.map_spanning_subgraphs GH) :=
+begin
+  induction p,
+  { simp only [transfer, map_nil], },
+  { simp only [p_ih, transfer, map_cons, hom.map_spanning_subgraphs_apply, eq_self_iff_true,
+             heq_iff_eq, and_self], },
+end
+
+@[simp] lemma edges_transfer : (p.transfer H hp).edges = p.edges :=
+begin
+  induction p,
+  { simp only [transfer, edges_nil], },
+  { simp only [p_ih, transfer, edges_cons, eq_self_iff_true, and_self], },
+end
+
+@[simp] lemma support_transfer : (p.transfer H hp).support = p.support :=
+begin
+  induction p,
+  { simp only [transfer, support_nil, eq_self_iff_true, and_self], },
+  { simp only [p_ih, transfer, support_cons, eq_self_iff_true, and_self], },
+end
+
+lemma is_path_transfer (pp : p.is_path) : (p.transfer H hp).is_path :=
+begin
+  induction p,
+  { simp only [transfer, is_path.nil], },
+  { simp only [cons_is_path_iff, transfer, support_transfer] at pp ⊢,
+  simp only [p_ih, pp, not_false_iff, and_self], },
+end
+
+lemma is_cycle_transfer {u : V} (p : G.walk u u) {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set) (pc : p.is_cycle) : (p.transfer H hp).is_cycle :=
+begin
+  cases p,
+  { simp only [transfer, is_cycle.not_of_nil] at pc ⊢, exact pc, },
+  { simp only [path.cons_is_cycle_iff, transfer, edges_transfer] at pc ⊢,
+    refine ⟨_, pc.right⟩,
+    apply is_path_transfer,
+    exact pc.left, },
+end
+
+lemma is_cycle_transfer_le {u : V} (p : G.walk u u) {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set) (pc : p.is_cycle)
+  (GH : G ≤ H) : (p.transfer H hp).is_cycle :=
+p.is_cycle_transfer (λ e ep, edge_set_mono GH (edges_subset_edge_set p ep)) pc
+
+lemma comp_transfer {K : simple_graph V} (hp' : ∀ e, e ∈ p.edges → e ∈ K.edge_set) :
+  (p.transfer H hp).transfer K (by { rw p.edges_transfer hp, exact hp', }) = p.transfer K hp' :=
+begin
+  induction p,
+  { simp only [transfer], },
+  { simp only [transfer, eq_self_iff_true, heq_iff_eq, true_and],
+   apply p_ih, },
```

Comment:
```suggestion
    apply p_ih }
```

Reply:
As for the commas: I like having commas everywhere: after each tactic call, *and* after each subblock. I don't think it makes sense to change just this one, but is my style here not acceptable/compatible with mathlib?

Reply:
You can keep the commas, but the indentation is important.